reporter.go

```diff
@@ -22,39 +22,88 @@ package tally
 
 import "time"
 
-// BaseStatsReporter implements the shared reporter methods
+// BaseStatsReporter implements the shared reporter methods.
 type BaseStatsReporter interface {
 	Capabilities() Capabilities
 	Flush()
 }
 
-// StatsReporter is a backend for Scopes to report metrics to
+// StatsReporter is a backend for Scopes to report metrics to.
 type StatsReporter interface {
 	BaseStatsReporter
 
 	// ReportCounter reports a counter value
-	ReportCounter(name string, tags map[string]string, value int64)
+	ReportCounter(
+		name string,
+		tags map[string]string,
+		value int64,
+	)
 
 	// ReportGauge reports a gauge value
-	ReportGauge(name string, tags map[string]string, value float64)
+	ReportGauge(
+		name string,
+		tags map[string]string,
+		value float64,
+	)
 
 	// ReportTimer reports a timer value
-	ReportTimer(name string, tags map[string]string, interval time.Duration)
+	ReportTimer(
+		name string,
+		tags map[string]string,
+		interval time.Duration,
+	)
+
+	// ReportHistogramValueSamples reports histogram samples for a bucket
+	ReportHistogramValueSamples(
+		name string,
+		tags map[string]string,
+		buckets Buckets,
+		bucketLowerBound,
+		bucketUpperBound float64,
+		samples int64,
+	)
+
+	// ReportHistogramDurationSamples reports histogram samples for a bucket
+	ReportHistogramDurationSamples(
+		name string,
+		tags map[string]string,
+		buckets Buckets,
+		bucketLowerBound,
+		bucketUpperBound time.Duration,
```

Comment:
Why do we have to pass the lower & upper bounds separately? Can't they be inferred from the buckets themselves? (Or be part of the buckets interface?)